src/transaction/txParser.ts

```diff
@@ -278,34 +283,64 @@ const parseMetaDataHash = (metaDataHash: any): Buffer | null => {
   return metaDataHash || null
 }
 
-const parseUnsignedTx = ([txBody, meta]: _UnsignedTxDecoded): _UnsignedTxParsed => {
-  if (txBody.get(TxBodyKeys.MINT)) {
-    throw Error(Errors.MintUnsupportedError)
+const deconstructUnsignedTxDecoded = (unsignedTxDecoded: any): _UnsignedTxDecoded => {
+  if (unsignedTxDecoded.length === 2) return unsignedTxDecoded
+  if (unsignedTxDecoded.length === 3) {
+    const [txBody, scriptWitnesses, metaData] = unsignedTxDecoded
+    if (scriptWitnesses != null && scriptWitnesses?.length > 0) {
+      throw Error(Errors.ScriptWitnessesNotSupported)
+    }
+    return [txBody, metaData]
   }
-  const inputs = parseTxInputs(txBody.get(TxBodyKeys.INPUTS))
-  const outputs = parseTxOutputs(txBody.get(TxBodyKeys.OUTPUTS))
-  const fee = parseFee(txBody.get(TxBodyKeys.FEE))
-  const ttl = parseTtl(txBody.get(TxBodyKeys.TTL))
-  const certificates = parseTxCerts(
-    txBody.get(TxBodyKeys.CERTIFICATES) || [],
-  )
-  const withdrawals = parseTxWithdrawals(
-    txBody.get(TxBodyKeys.WITHDRAWALS) || new Map(),
-  )
-  const metaDataHash = parseMetaDataHash(txBody.get(TxBodyKeys.META_DATA_HASH))
-  const validityIntervalStart = parseValidityIntervalStart(txBody.get(TxBodyKeys.VALIDITY_INTERVAL_START))
+  throw Error(Errors.FailedToParseTransaction)
+}
 
-  return {
-    inputs,
-    outputs,
-    fee,
-    ttl,
-    certificates,
-    withdrawals,
-    metaDataHash,
-    meta,
-    validityIntervalStart,
-    mint: null, // unsupported in current version
+const parseUnsignedTx = (unsignedTxCborHex: UnsignedTxCborHex): _UnsignedTxParsed => {
+  const originalTxDecoded = decodeCbor(unsignedTxCborHex)
+  const unsignedTxDecoded = deconstructUnsignedTxDecoded(originalTxDecoded)
+  if (!isUnsignedTxDecoded(unsignedTxDecoded)) {
+    throw Error(Errors.InvalidTransactionBody)
+  } else {
+    const [txBody, meta] = unsignedTxDecoded
+    if (txBody.get(TxBodyKeys.MINT)) {
+      throw Error(Errors.MintUnsupportedError)
+    }
+    const inputs = parseTxInputs(txBody.get(TxBodyKeys.INPUTS))
+    const outputs = parseTxOutputs(txBody.get(TxBodyKeys.OUTPUTS))
+    const fee = parseFee(txBody.get(TxBodyKeys.FEE))
+    const ttl = parseTtl(txBody.get(TxBodyKeys.TTL))
+    const certificates = parseTxCerts(
+      txBody.get(TxBodyKeys.CERTIFICATES) || [],
+    )
+    const withdrawals = parseTxWithdrawals(
+      txBody.get(TxBodyKeys.WITHDRAWALS) || new Map(),
+    )
+    const metaDataHash = parseMetaDataHash(txBody.get(TxBodyKeys.META_DATA_HASH))
+    const validityIntervalStart = parseValidityIntervalStart(txBody.get(TxBodyKeys.VALIDITY_INTERVAL_START))
+
+    const getId = (): string => {
```

Comment:
I think we want a unittest checking that getId of a parsed unsigned tx corresponds to the hash of original cbor-encoded tx. I think there is one, though I'm not 100% sure from the code structure.

Reply:
Yes, there are unit tests for that already.